hud.c

```diff
@@ -912,19 +926,87 @@ static const ItemBoxGfx *Hud_GetIconForItem(int i) {
 }
 
+void CopyTilesForSwitchLR(int switch_lr) {
+#define PV(a0,a1,a2,a3,a4,a5,a6,a7)  ((a0 & 1) << 7 | (a0 >> 1 & 1) << 15 | (a1 & 1) << 6 | (a1 >> 1 & 1) << 14 | (a2 & 1) <<5 | (a2 >> 1&1) <<13 | (a3 & 1) << 4 | (a3>> 1 & 1) << 12 | (a4 & 1) << 3 | (a4 >> 1 & 1) << 11 | (a5 & 1) << 2 | (a5 >> 1 & 1) << 10 | (a6 & 1) << 1 | (a6 >> 1 & 1) << 9 | (a7 & 1) << 0 | (a7 >> 1 & 1) << 8) 
+    
+  if (switch_lr == 2) {
+    static const uint16 kBytesForNewTile0xC[8] = {
+      PV(1,1,1,1,1,1,3,3),
+      PV(1,1,1,1,1,1,1,3),
+      PV(1,1,1,1,1,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,1,1,1,1,3)
+    };
+    memcpy(&g_zenv.vram[0x7000 + 0xc * 8], kBytesForNewTile0xC, sizeof(kBytesForNewTile0xC));
+
+    static const uint16 kBytesForNewTile0xD[8] = {
+      PV(1,1,1,1,1,1,3,3),
+      PV(1,1,1,3,1,1,1,3),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1),
+      PV(1,1,1,3,3,1,1,1)
+    };
+    memcpy(&g_zenv.vram[0x7000 + 0xd * 8], kBytesForNewTile0xD, sizeof(kBytesForNewTile0xD));;
+  }
+
+  if (switch_lr == 3) {
+    static const uint16 kBytesForNewTile0xE[8] = {
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3)
+    };
+    memcpy(&g_zenv.vram[0x7000 + 0xe * 8], kBytesForNewTile0xE, sizeof(kBytesForNewTile0xE));
+
+    static const uint16 kBytesForNewTile0xF[8] = {
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,3,3,3,3,3),
+      PV(1,1,1,1,1,1,1,1),
+      PV(1,1,1,1,1,1,1,1),
+      PV(1,1,1,1,1,1,1,1)
+    };
+    memcpy(&g_zenv.vram[0x7000 + 0xf * 8], kBytesForNewTile0xF, sizeof(kBytesForNewTile0xF)); 
+  }
+
+}
+
+static const uint8 kSwitchLR_palettes[] = { 7, 3, 4,4 };
+
 void Hud_DrawYButtonItems() {  // 8de3d9
   uint16 *dst = uvram_screen.row[0].col;
   int x = kNewStyleInventory ? 0 : 1;
-
-  bool is_x = (joypad1L_last & kJoypadL_X && (enhanced_features0 & kFeatures0_SwitchLR));
-  uint8 palette = is_x ? 3 : 7;
-  Hud_DrawBox(dst, x, 5, 20 - x, 19, palette);
+  
+  CopyTilesForSwitchLR(2);
+  CopyTilesForSwitchLR(3);
+  Hud_DrawBox(dst, x, 5, 20 - x, 19, kSwitchLR_palettes[GetCurrentItemButtonIndex()]);
 
   if (!kNewStyleInventory) {
-    if (palette == 3) {
+    if (GetCurrentItemButtonIndex() == 1) {
       dst[HUDXY(2, 6)] = 0x2CF0;
       dst[HUDXY(2, 7)] = 0x2CF0 | 0x8000;
-    } else {
+    } 
+    else if (GetCurrentItemButtonIndex() == 2) {
+      dst[HUDXY(2, 6)] = 0x200E | kSwitchLR_palettes[GetCurrentItemButtonIndex()] << 10;
+      dst[HUDXY(2, 7)] = 0x200F | kSwitchLR_palettes[GetCurrentItemButtonIndex()] << 10;
+    }
+    else if (GetCurrentItemButtonIndex() == 3) {

```

Comment:
this gets too messy with all those ifs now that there's so many
change to
static const uint16 kEquipmentItemTiles[4][2] = {
  {0x3cf0, 0x3cf1},  // Y
  {0x2CF0, 0x2CF0 | 0x8000},  // X
 ...
};
dst[HUDXY(2, 6)] = kEquipmentItemTiles[itembutton][0];
dst[HUDXY(2, 7)] = kEquipmentItemTiles[itembutton][1];